File: crates/dewasm-backend-go/units/wasi/fd_sync.go
func (w *WASI) wasi_fd_sync(fd uint32) uint32 {
    f, ok := w.fds[fd].(*os.File)
    if !ok {
        return wasiBadf
    }
    if err := f.Sync(); err != nil {
        return wasiIo
    }
    return wasiOk
}

File: crates/dewasm-backend-java/units/wasi/fd_sync.java
int wasi_fd_sync(int fd) {
    Object e = fds.get(fd);
    if (!(e instanceof Handle)) {
        return WASI_BADF;
    }
    try {
        ((Handle) e).ch.force(true);
    } catch (java.io.IOException ex) {
        return WASI_IO;
    }
    return WASI_OK;
}
